### PR TITLE
Allow disabling errors

### PR DESCRIPTION
**Pull request by pfistfl**
Introduces a new arg `friedman_global` that allows disabling errors if the friedman global test does not pass for critical differences plots.

### CODE REVIEW

**Comment by pfistfl** [comment]
Solves #21

File: R/BenchmarkAggr.R [hunk 1]
```diff
@@ -60,6 +60,7 @@ BenchmarkAggr = R6Class("BenchmarkAggr",
       private$.independent = assert_flag(independent)
       assert_flag(strip_prefix)
       assert_subset(c(task_id, learner_id), colnames(dt))
+      dt = map_at(dt, c(task_id, learner_id), as.factor)
```
Comment (RaphaelS1) [inline R/BenchmarkAggr.R:63]: I think we removed automatically converting to a factor as it easily leads to errors

File: R/BenchmarkAggr.R [hunk 2]
```diff
@@ -324,8 +325,12 @@ BenchmarkAggr = R6Class("BenchmarkAggr",
 
       # Perform nemenyi test
       nem_test = tryCatch(self$friedman_posthoc(meas, p.value),
-                   warning = function(w)
-                     stopf("Global Friedman test non-significant (p > %s), try type = 'mean' instead.", p.value)) # nolint
+                 warning = function(w) {
+                   if (friedman_posthoc)
+                     stopf("Global Friedman test non-significant (p > %s), try type = 'mean' instead.", p.value) # nolint
+                   else
+                     warning(sprintf("Global Friedman test non-significant (p > %s), try type = 'mean' instead.", p.value)) # nolint))
```
Comment (RaphaelS1) [inline R/BenchmarkAggr.R:332]: I don't think we need a warning here, if `friedman_posthoc=FALSE` as default then the user should know what they're doing when they change it

File: R/autoplot.BenchmarkAggr.R [hunk 3]
```diff
@@ -44,6 +44,10 @@
 #' specifying the aspect ratio of the plot, best used with [ggsave()].
 #' @param col (`character(1)`)\cr
 #' For `type = "fn"`, specifies color to fill significant tiles, default is `"red"`.
+#' @param friedman_global (`logical(1)`)\cr
+#' Should a friedman global test be performed for`type = "cd"` and `type = "fn"`?
+#' If `FALSE`, a warning is issued in case the corresponding friedman posthoc test fails instead of an error.
+#' Default is `TRUE` (raises an error).
```
Comment (RaphaelS1) [inline R/autoplot.BenchmarkAggr.R:50]:
```suggestion
#' Default is `TRUE` (raises an error if global test fails).
```

File: R/autoplot.BenchmarkAggr.R [hunk 4]
```diff
@@ -108,15 +112,21 @@ autoplot.BenchmarkAggr = function(obj, type = c("mean", "box", "fn", "cd"), meas
...
-    p = p[rev(seq_len(nrow(p))), ]
-    p = t(p)
-
-    p = cbind(expand.grid(rownames(p), colnames(p)), value = as.numeric(p))
-
+                 warning = function(w) {
+                   if (friedman_global)
+                     stopf("Global Friedman test non-significant (p > %s), try type = 'mean' instead.", p.value) # nolint
+                   else
+                     warning(sprintf("Global Friedman test non-significant (p > %s), try type = 'mean' instead.", p.value)) # nolint))
```
Comment (RaphaelS1) [inline R/autoplot.BenchmarkAggr.R:119]: As above, I don't think the warning is needed

File: R/autoplot.BenchmarkAggr.R [hunk 5]
```diff
@@ -108,15 +112,21 @@ autoplot.BenchmarkAggr = function(obj, type = c("mean", "box", "fn", "cd"), meas
...
+                 })
+    if (friedman_global) {
+      p = p[rev(seq_len(nrow(p))), ]
+      p = t(p)
+      p = cbind(expand.grid(rownames(p), colnames(p)), value = as.numeric(p))
+    } else {
+      p = cbind(expand.grid(obj$data[[obj$col_roles$learner_id]],  obj$data[[obj$col_roles$task_id]]), value = 1)
```
Comment (RaphaelS1) [inline R/autoplot.BenchmarkAggr.R:126]: Shouldn't `p` always equal the result of the posthoc?

File: R/BenchmarkAggr.R [hunk 6]
```diff
@@ -298,7 +299,7 @@ BenchmarkAggr = R6Class("BenchmarkAggr",
     .dt = data.table(),
     .independent = logical(0),
     .crit_differences = function(meas = NULL, minimize = TRUE, p.value = 0.05, baseline = NULL, # nolint
-                                test = c("bd", "nemenyi")) {
+                                test = c("bd", "nemenyi"), friedman_posthoc = TRUE) {
```
Comment (RaphaelS1) [inline R/BenchmarkAggr.R:302]: Seems confusingly named, shouldn't this be friedman_global?

**Review by RaphaelS1 — CHANGES_REQUESTED** [review]
Thanks for this, some small comments

File: R/BenchmarkAggr.R (hunk 1 above)
Comment (RaphaelS1) [reply R/BenchmarkAggr.R:63]:
```suggestion
```

**Pull request merged**